test/unit-test/FreeRTOS_Sockets/FreeRTOS_Sockets_utest.c

```diff
@@ -151,3 +218,7366 @@ void test_FreeRTOS_inet_pton4( void )
     TEST_ASSERT_EQUAL( pdFAIL, xResult );
     TEST_ASSERT_EQUAL_MESSAGE( ulInValidResponse, ulIPAddress, "Incorrectly converted string 8." );
 }
+
+void test_prvValidSocket_InvalidOrNULLSocket( void )
+{
+    BaseType_t xReturn;
+    FreeRTOS_Socket_t * pxSocket = ( FreeRTOS_Socket_t * ) FREERTOS_INVALID_SOCKET;
+    BaseType_t xProtocol, xIsBound;
+
+    xReturn = prvValidSocket( NULL, xProtocol, xIsBound );
+
+    TEST_ASSERT_EQUAL( pdFALSE, xReturn );
+
+    xReturn = prvValidSocket( pxSocket, xProtocol, xIsBound );
+
+    TEST_ASSERT_EQUAL( pdFALSE, xReturn );
+}
+
+void test_prvValidSocket_SocketBoundSetButNotBound( void )
+{
+    BaseType_t xReturn;
+    FreeRTOS_Socket_t xSocket;
+    BaseType_t xProtocol, xIsBound = pdTRUE;
+
+    memset( &xSocket, 0, sizeof( FreeRTOS_Socket_t ) );
+
+    listLIST_ITEM_CONTAINER_ExpectAndReturn( &( xSocket.xBoundSocketListItem ), NULL );
+
+    xReturn = prvValidSocket( &xSocket, xProtocol, xIsBound );
+
+    TEST_ASSERT_EQUAL( pdFALSE, xReturn );
+}
+
+void test_prvValidSocket_SocketBoundResetButBound( void )
+{
+    BaseType_t xReturn;
+    FreeRTOS_Socket_t xSocket;
+    BaseType_t xProtocol, xIsBound = pdFALSE;
+
+    memset( &xSocket, 0, sizeof( FreeRTOS_Socket_t ) );
+
+    xSocket.ucProtocol = xProtocol;
+
+    xReturn = prvValidSocket( &xSocket, xProtocol, xIsBound );
+
+    TEST_ASSERT_EQUAL( pdTRUE, xReturn );
+}
+
+void test_prvValidSocket_InvalidProtocol( void )
+{
+    BaseType_t xReturn;
+    FreeRTOS_Socket_t xSocket;
+    BaseType_t xProtocol, xIsBound = pdTRUE;
+
+    memset( &xSocket, 0, sizeof( FreeRTOS_Socket_t ) );
+
+    listLIST_ITEM_CONTAINER_ExpectAndReturn( &( xSocket.xBoundSocketListItem ), ( struct xLIST * ) ( uintptr_t ) 0x11223344 );
+
+    xSocket.ucProtocol = xProtocol + 1;
+
+    xReturn = prvValidSocket( &xSocket, xProtocol, xIsBound );
+
+    TEST_ASSERT_EQUAL( pdFALSE, xReturn );
+}
+
+void test_vNetworkSocketsInit( void )
+{
+    vListInitialise_Expect( &xBoundUDPSocketsList );
+    vListInitialise_Expect( &xBoundTCPSocketsList );
+
+    vNetworkSocketsInit();
+}
+
+void test_prvDetermineSocketSize_IPTaskNotInit( void )
+{
+    BaseType_t xReturn;
+    BaseType_t xDomain, xType, xProtocol;
+    size_t xSocketSize;
+
+    xIPIsNetworkTaskReady_ExpectAndReturn( pdFALSE );
+
+    xReturn = prvDetermineSocketSize( xDomain, xType, xProtocol, &xSocketSize );
+
+    TEST_ASSERT_EQUAL( pdFAIL, xReturn );
+}
+
+void test_prvDetermineSocketSize_CatchAssert( void )
```

Comment:
Which assertion are you catching here?

Reply:
That the domain should always be `FREERTOS_AF_INET`.

Reply:
Could you adjust the same of this test case to make that clear or add a comment?

Reply:
done